app/components/bali/hover_card/component.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Bali
+  module HoverCard
+    class Component < ApplicationViewComponent
+      attr_reader :placement, :options, :data, :hover_url
+
+      renders_one :trigger, ->(**options, &block) do
+        options = prepend_class_name(options, 'hovercard-component--trigger')
+
+        tag.div(**options, &block)
+      end
+
+      renders_one :template
+
+      def initialize(hover_url: nil, placement: 'auto', **options)
+        @placement = placement
+        @data = options.delete(:data) || {}
+        @hover_url = options.dig(:data, :'hovercard-url-value') || hover_url
+        @options = prepend_class_name(options, 'hover-card-component')
+      end
+
+      private
+
+      def hovercard_data
+        {
+          controller: "hovercard #{data.delete(:controller)}",
+          'hovercard-placement-value': placement,
+          'hovercard-url-value': hover_url
+        }.merge!(data).compact
+      end
```

Comment:
De esta manera ya puedes quitar el hovercard_data del template y solamente pasarle las options.

```suggestion
        @options = prepend_class_name(options, 'hover-card-component')
        @options = prepend_controller(@options, 'hovercard')
        @options = prepend_data_attribute(@options, 'hovercard-placement-value', placement)
        @options = prepend_data_attribute(@options, 'hovercard-url-value', hover_url)
      end
```